src/problem.jl

```diff
@@ -0,0 +1,53 @@
+abstract type AbstractNeuralNetworkControlProblem end
+
+# ST: type of system
+# XT: type of initial condition
+# DT: type of variables
+# PT: type of period
+struct ControlledPlant{ST, XT, DT, PT} <: AbstractNeuralNetworkControlProblem
+    ivp::InitialValueProblem{ST, XT}
+    controller::Network
+    vars::Dict{Symbol, DT}
+    period::PT
+end
+
+plant(cp::ControlledPlant) = cp.ivp
+system(cp::ControlledPlant) = cp.ivp.s
+controller(cp::ControlledPlant) = cp.controller
+period(cp::ControlledPlant) = cp.period
+
+function state_vars(cp::ControlledPlant)
```

Comment:
unrelated to the PR, but i would just write this in one line: `state_vars(cp::...) = get(throw(ArgumentError("...")), cp.vars, :state_vars)`

Reply:
I just copy-pasted them (and added some other code). I prefer to postpone this after merging.